test_launch_ros/test/test_launch_ros/actions/test_set_parameters_from_file.py

```diff
@@ -0,0 +1,136 @@
+    lc = MockContext()
+    node_1 = Node(
+        package='asd',
+        executable='bsd',
+        name='my_node',
+        namespace='my_ns',
+        parameters=[{'asd': 'bsd1'}]
+    )
+
+    node_2 = Node(
+        package='asd',
+        executable='bsd',
+        name='my_other_node',
+        namespace='my_ns',
+        parameters=[{'asd': 'bsd2'}]
+    )
+    param_file_path = os.path.dirname(os.path.abspath(__file__)) + '/example_parameters_1.yaml'
+    set_param_file = SetParametersFromFile(param_file_path)
+    set_param_single = SetParameter(name='my_param', value='my_value')
+    set_param_file.execute(lc)
+    set_param_single.execute(lc)
+    node_1._perform_substitutions(lc)
+    node_2._perform_substitutions(lc)
+
+    actual_command_1 = [perform_substitutions(lc, item) for item in
+                        node_1.cmd if type(item[0]) == TextSubstitution]
+
+    actual_command_2 = [perform_substitutions(lc, item) for item in
+                        node_2.cmd if type(item[0]) == TextSubstitution]
+
+    assert actual_command_1[3] == '--params-file'
+    assert os.path.isfile(actual_command_1[4])
+    assert actual_command_1[5] == '-p'
+    assert actual_command_1[6] == 'my_param:=my_value'
+    assert actual_command_1[7] == '--params-file'
+    assert os.path.isfile(actual_command_1[8])
+
+    expected_param_file = {
+        '/**': {
+            'ros__parameters': {'param_1_name': 10}
+        },
+        '/my_ns': {
+            'my_node': {
+                'ros__parameters': {'param_2_name': 20}
+            },
+            'my_other_node': {
+                'ros__parameters': {'param_3_name': 30}
+            }
+        }
+    }
+
+    with open(actual_command_1[4], 'r') as h:
+        expanded_parameters_dict = yaml.load(h, Loader=yaml.FullLoader)
+        assert expanded_parameters_dict == expected_param_file
+
+    with open(actual_command_1[8], 'r') as h:
+        expanded_parameters_dict = yaml.load(h, Loader=yaml.FullLoader)
+        assert expanded_parameters_dict == {
+            '/my_ns/my_node': {
+                'ros__parameters': {'asd': 'bsd1'}
+            }
+        }
+
+    assert actual_command_2[3] == '--params-file'
+    assert os.path.isfile(actual_command_2[4])
+    assert actual_command_2[5] == '-p'
+    assert actual_command_2[6] == 'my_param:=my_value'
+    assert actual_command_2[7] == '--params-file'
+    assert os.path.isfile(actual_command_2[8])
+
+    with open(actual_command_2[4], 'r') as h:
+        expanded_parameters_dict = yaml.load(h, Loader=yaml.FullLoader)
+        assert expanded_parameters_dict == expected_param_file
+
+    with open(actual_command_2[8], 'r') as h:
+        expanded_parameters_dict = yaml.load(h, Loader=yaml.FullLoader)
+        assert expanded_parameters_dict == {
+            '/my_ns/my_other_node': {
+                'ros__parameters': {'asd': 'bsd2'}
+            }
+        }
+
+# TODO: (Aditya) Add test case for composable node
```

Comment:
it would be great to add that before merging

Reply:
```SetParametersFromFile``` currently only works for nodes and not composable nodes.
As composable nodes and nodes currently use different styles for accepting parameters from yaml files, https://github.com/ros2/launch_ros/pull/259 would need to be closed before I can add that functionality to composable nodes. We could:

1. Add a note to ```SetParametersFromFile``` saying that this won't affect composable nodes, and close this PR. Implement it in a new PR once #259 is merged.
2. Wait for #259 to be merged.

Personally, I think we can proceed with option 1. How should we proceed ? @ivanpauno

Reply:
Yes, merging this and iterating after https://github.com/ros2/launch_ros/pull/259 is finished sounds good to me.